pallets/claim/src/lib.rs

```diff
@@ -164,89 +180,172 @@ decl_error! {
 			let who = ensure_signed(origin)?;
 			let relayer = Relayer::<T>::get();
 			ensure!(relayer.is_some(), Error::<T>::NoRelayer);
-			ensure!(Some(&who) == relayer.as_ref(), Error::<T>::CallerNotRelayer);
+			ensure!(
+				Some(&who) == relayer.unwrap().as_ref(),
+				Error::<T>::CallerNotRelayer
+			);
 			// check first
 			for (eth_tx_hash, _, _) in claims.iter() {
-				ensure!(!BurnedTransactions::<T>::contains_key(&eth_tx_hash), Error::<T>::TxHashAlreadyExist);
+				ensure!(
+					!BurnedTransactions::<T>::contains_key(&eth_tx_hash),
+					Error::<T>::TxHashAlreadyExist
+				);
 			}
 			for (eth_tx_hash, eth_address, erc20_amount) in claims.iter() {
-				BurnedTransactions::<T>::insert(&eth_tx_hash, (eth_address.clone(), erc20_amount.clone()));
-				ClaimState::insert(&eth_tx_hash, false);
-				Self::deposit_event(RawEvent::ERC20TransactionStored(who.clone(), *eth_tx_hash, *eth_address, *erc20_amount));
+				BurnedTransactions::<T>::insert(
+					&eth_tx_hash,
+					(eth_address.clone(), erc20_amount.clone()),
+				);
+				ClaimState::<T>::insert(&eth_tx_hash, false);
+				Self::deposit_event(Event::ERC20TransactionStored(
+					who.clone(),
+					*eth_tx_hash,
+					*eth_address,
+					*erc20_amount,
+				));
 			}
-			let end_height = EndHeight::get();
-			   if height > end_height {
-				EndHeight::put(height);
+			if let Some(end_height) = EndHeight::<T>::get() {
+				if height > end_height {
+					EndHeight::<T>::put(height);
+				}
+			} else {
+				EndHeight::<T>::put(height);
 			}
-			Ok(())
+			Ok(().into())
 		}
 
-		#[weight = 0]
-		pub fn claim_erc20_token(origin, account: T::AccountId, eth_tx_hash: EthereumTxHash, eth_signature: EcdsaSignature) -> dispatch::DispatchResult {
+		#[pallet::weight(0 + T::DbWeight::get().reads_writes(1,1))]
+		pub fn claim_erc20_token(
+			origin: OriginFor<T>,
+			account: T::AccountId,
+			eth_tx_hash: EthereumTxHash,
+			eth_signature: EcdsaSignature,
+		) -> DispatchResultWithPostInfo {
 			let _ = ensure_none(origin)?;
-			ensure!(BurnedTransactions::<T>::contains_key(&eth_tx_hash), Error::<T>::TxHashNotFound);
-			ensure!(!ClaimState::get(&eth_tx_hash), Error::<T>::TxAlreadyClaimed);
+			ensure!(
+				BurnedTransactions::<T>::contains_key(&eth_tx_hash),
+				Error::<T>::TxHashNotFound
+			);
+			ensure!(
+				ClaimState::<T>::get(&eth_tx_hash).is_some(),
+				Error::<T>::TxAlreadyClaimed
+			);
 			let address = Encode::encode(&account);
 			let signer = Self::eth_recover(&eth_signature, &address, &eth_tx_hash.0)
 				.ok_or(Error::<T>::InvalidSignature)?;
-			let tx = BurnedTransactions::<T>::get(&eth_tx_hash);
-			ensure!(signer == tx.0, Error::<T>::InvalidSigner);
-			ClaimState::insert(&eth_tx_hash, true);
-			// mint coins
-			let imbalance = T::Currency::deposit_creating(&account, tx.1);
-			drop(imbalance);
-			Self::deposit_event(RawEvent::ERC20TokenClaimed(account, eth_tx_hash, tx.1));
-			Ok(())
+			if let Some(tx) = BurnedTransactions::<T>::get(&eth_tx_hash) {
```

Comment:
Again, unwrap_or_default(). I didn't check if the default value can cause different behavior here